src/buffers.h

```diff
@@ -0,0 +1,18 @@
+#ifndef SRC_BUFFERS_H_
+#define SRC_BUFFERS_H_
+
+// Translated into UTF-8, this means that a worst-case UTF-16
+// buffer would be double (4 bytes).
+#define MAX_UTF8_BUFFER 256
+
+// We frequently have terminator characters with the UTF-8 buffers.
+#define MAX_UTF8_BUFFER_WITH_TERM (MAX_UTF8_BUFFER + 1)
+
+// When converting UTF-16 to UTF-8, we still have 1-byte characters
+// but it can double the length when going from a wide to two bytes.
+#define MAX_UTF16_TO_UTF8_BUFFER MAX_UTF8_BUFFER
+
+// Converting between buffers needs a bit of space also.
+#define MAX_UTF8_TO_UTF8_BUFFER (MAX_UTF8_BUFFER * 2 + 1)
```

Comment:
Correct me if I'm wrong, but it seems like this constant name is a bit misleading? You're using it when converting between arbitrary 8-byte encodings. Would a better name be `MAX_8_TO_8_BUFFER`?

Reply:
I changed it to `MAX_TRANSCODE_BUFFER` to hopefully make it more clear what we're doing with it. I don't know if the byte name was appropriate since the *2 was because of the UTF-8 to UTF-16 to UTF-8 loop that Windows requires.

Reply:
I like your new name a lot.